Wizegolangapi

MiguelAGrover opened the pull request:
This Pull request include all the changes for the final delivery of the bootcamp.

Please take a look.

MiguelAGrover committed:
Add functional route for create a digimon into the database

Create the functional route for  create a digimon into the csv, also change the database to manipulate a csv

MiguelAGrover committed:
Added update method of Digimon

Added route to update a Digimon in the CSV.

MiguelAGrover committed:
Added update and delete routes

Added update and delete routes for digimon Data

MiguelAGrover committed:
Added testcases and modify comments

Added testcases for csvdb and changed comments on some files to be more descriptive

MiguelAGrover committed:
Added comments and change find logic

Change comments to be more readable and change find logic for digimon repository to implement dictionaries

MiguelAGrover committed:
Added testcases for digimon repository

MiguelAGrover committed:
Created service for external APIs

Created services for request to external APIs

MiguelAGrover committed:
Made corrections as suggested by mentor

MiguelAGrover committed:
Added testcases for service package

alvarocueva commented on infraestructure/datastore/csvdb.go:
```diff
@@ -0,0 +1,150 @@
+	"time"
+)
+
+// MyError : the structure that contain personalized error with information related to read and write to csvs
```
The comment should start with the struct name `CSVDBError`

alvarocueva commented on infraestructure/datastore/csvdb_test.go:
```diff
@@ -0,0 +1,64 @@
+)
+
+func Test_ReadCSV(t *testing.T) {
+	db := NewCSVDB("../../test_utilities/test_data.csv")
```
This path is used more than 2 times, do it a constant

alvarocueva commented on interface/repository/digimon_repository.go:
```diff
@@ -1,32 +1,196 @@
-
-	"gorm.io/gorm"
+	"errors"
+	"fmt"
```
Separate the imports

alvarocueva commented on interface/repository/digimon_repository.go:
```diff
@@ -1,32 +1,196 @@
+	for index, rec := range data {
+		if index == 0 {
+			labels = rec
+			digimonmap = make(map[string]string)
```
This line is not necessary, at least not here

alvarocueva commented on interface/repository/digimon_repository.go:
```diff
@@ -1,32 +1,196 @@
+			digimonmap[labels[i]] = value
+		}
+		digimon := model.Digimon{
+			Name:  digimonmap["Name"],
```
`"Name` is used more than 2 times, declare it as a constant

alvarocueva commented on interface/repository/digimon_repository.go:
```diff
@@ -1,32 +1,196 @@
+	for index, rec := range data {
+		if index == 0 {
+			labels = rec
+			digimonmap = make(map[string]string)
```
This line is not necessary, at least not here

alvarocueva commented on interface/repository/digimon_repository_test.go:
```diff
@@ -0,0 +1,77 @@
+
+import (
+	"testing"
+	"wizegolangapi/domain/model"
```
Separate the imports

alvarocueva commented on interface/repository/digimon_repository_test.go:
```diff
@@ -0,0 +1,77 @@
+)
+
+func Test_CreateDigimon(t *testing.T) {
+	db := datastore.NewCSVDB("../../test_utilities/test_data.csv")
```
This path is used more than 2 times, declare it as a constant

alvarocueva commented on interface/repository/digimon_repository_test.go:
```diff
@@ -0,0 +1,77 @@
+func Test_CreateDigimon(t *testing.T) {
+	db := datastore.NewCSVDB("../../test_utilities/test_data.csv")
+	dr := NewDigimonRepository(db)
+	digimon := &model.Digimon{"Test", "Test", "Test"}
```
`Test` more than 2 times -> constant

MiguelAGrover committed:
Constants creation and error in comments

Change name on comment to correct error name and added constants for strings that appear more than once.
